test/controllers/thesis_controller_test.rb

```diff
@@ -353,4 +353,35 @@ class ThesisControllerTest < ActionDispatch::IntegrationTest
     assert_equal path, thesis_process_path(theses(:one))
   end
 
+  test 'removing a file from a thesis will be reflected with a link to its transfer in the flash message' do
```

Comment:
An additional test where multiple files are attached and only one is removed and the rest remain would be helpful I think.

Reply:
I've ended up going with three tests, at this point:
1. Confirm that, when the form is submitted with no deletions, that no files are affected (measured by various counts)
2. Confirm that, when the form is submitted with a deletion, that the counts of records/files/blobs change in the way we expect
3. Confirm that, when the form is submitted with a deletion that the flash message afterwards contains a link back to the transfer record.

Along the way I'm adding a setup method for these three tests, similar to what I did with the transfer controller tests with commit 72bafc8 .

I think this makes this ready for re-review, but if I'm missing a test scenario I'm open to writing more.